homeassistant/components/tuya/light.py

```diff
@@ -565,27 +580,27 @@ def turn_on(self, **kwargs: Any) -> None:
                     },
                 ]
 
-            commands += [
-                {
-                    "code": self._color_temp.dpcode,
-                    "value": round(
-                        self._color_temp.remap_value_from(
-                            color_util.color_temperature_kelvin_to_mired(
-                                kwargs[ATTR_COLOR_TEMP_KELVIN]
-                            ),
-                            MIN_MIREDS,
-                            MAX_MIREDS,
-                            reverse=True,
-                        )
-                    ),
-                },
-            ]
+            if ATTR_COLOR_TEMP_KELVIN in kwargs and self._color_temp:
+                commands += [
+                    {
+                        "code": self._color_temp.dpcode,
+                        "value": round(
+                            self._color_temp.remap_value_from(
+                                kwargs[ATTR_COLOR_TEMP_KELVIN],
+                                self.min_mireds,
+                                self.max_mireds,
+                                reverse=True,
+                            )
```

Comment:
This part of the code can't be correct. We are mapping kelvin to a min/max mireds here.